Shared/Samples/Orbit camera around object/OrbitCameraAroundObjectView.swift

```diff
@@ -0,0 +1,177 @@
+    var body: some View {
+        SceneView(
+            scene: model.scene,
+            cameraController: model.cameraController,
+            graphicsOverlays: [model.graphicsOverlay]
+        )
+        .disabled(sceneIsDisabled)
+        .toolbar {
+            ToolbarItemGroup(placement: .bottomBar) {
+                cameraViewPicker
+                settingsButton
+            }
+        }
+        .errorAlert(presentingError: $error)
+    }
+    
+    /// The picker for selecting the camera view.
+    @ViewBuilder private var cameraViewPicker: some View {
+        Picker("Camera View", selection: $selectedCameraView) {
+            Text("Center").tag(CameraView.center)
+            Text("Cockpit").tag(CameraView.cockpit)
+        }
+        .pickerStyle(.segmented)
+        .task(id: selectedCameraView) {
+            // Move the camera to the new view selection.
+            do {
+                // Disable scene interaction while the camera is moving.
+                sceneIsDisabled = true
+                defer { sceneIsDisabled = false }
+                
+                switch selectedCameraView {
+                case .center:
+                    try await model.moveToPlaneView()
+                case .cockpit:
+                    try await model.moveToCockpit()
+                }
+            } catch {
+                self.error = error
+            }
+        }
+    }
+    
+    /// The button that brings up the settings sheet.
+    @ViewBuilder private var settingsButton: some View {
+        let button = Button("Settings") {
+            settingsSheetIsPresented = true
+        }
+        let settingsContent = SettingsView(model: model)
+        
+        if #available(iOS 16, *) {
+            button
+                .popover(isPresented: $settingsSheetIsPresented, arrowEdge: .bottom) {
+                    settingsContent
+                        .presentationDetents([.fraction(0.5)])
+#if targetEnvironment(macCatalyst)
+                        .frame(minWidth: 300, minHeight: 270)
+#else
+                        .frame(minWidth: 320, minHeight: 390)
+#endif
+                }
+        } else {
+            button
+                .sheet(isPresented: $settingsSheetIsPresented, detents: [.medium]) {
+                    settingsContent
+                }
+        }
+    }
+}
+
+private extension OrbitCameraAroundObjectView {
+    /// The camera and plane settings for the sample.
+    struct SettingsView: View {
+        /// The view model for the sample.
+        @ObservedObject var model: Model
+        
+        /// The action to dismiss the view.
+        @Environment(\.dismiss) private var dismiss: DismissAction
+        
+        /// The pitch of the plane in the scene.
+        @State private var planePitch = Measurement(value: 0, unit: UnitAngle.degrees)
+        
+        /// The heading offset of the camera controller.
+        @State private var cameraHeading = Measurement(value: 0, unit: UnitAngle.degrees)
```

Comment:
Up to you, my personal preference is to use type inference
```suggestion
        @State private var planePitch = Measurement<UnitAngle>(value: 0, unit: .degrees)
        
        /// The heading offset of the camera controller.
        @State private var cameraHeading = Measurement<UnitAngle>(value: 0, unit: .degrees)
```